command/agent/check_test.go

```diff
@@ -226,9 +226,23 @@ func TestCheckHTTP(t *testing.T) {
 					w.WriteHeader(999)
 					return
 				}
-				if len(tt.header) > 0 && !reflect.DeepEqual(tt.header, r.Header) {
-					w.WriteHeader(999)
-					return
+				if len(tt.header) > 0 {
+					// Add the default headers to what was
```

Comment:
I think the comment can be shorter, e.g. `Merge default and custom headers`

Also, shouldn't the default headers always be there?

```go
h := http.Header{...}
for k, v := range tt.header {
    h[k] = v
}
if !reflect.DeepEqual(h, r.Header) { ...}
```